libraries/core_libs/consensus/include/dag/dag.hpp

```diff
@@ -165,6 +168,16 @@ class DagManager : public std::enable_shared_from_this<DagManager> {
 
   const std::pair<uint64_t, std::map<uint64_t, std::vector<blk_hash_t>>> getNonFinalizedBlocks() const;
 
+  /**
+   * @brief Sends non finalized dag blocks excluding the known block to the peer
+   *
+   * @param blocks_hashes Blocks known to the peer
+   * @param peer Peer to send the blocks to
+   * @param peer_period Peer period
+   */
+  void sendNonFinalizedBlocks(std::unordered_set<blk_hash_t> &&blocks_hashes,
+                              const std::shared_ptr<network::tarcap::TaraxaPeer> &peer, uint64_t peer_period);
```

Comment:
I think, we should not pass peer object, but return something like `getNonFinalizedBlocks`, but instead of hashes we return blocks for sending and period

Reply:
Same as the comment I left for Jakub. In that case it is not under the mutex. I am open for discussion on what would be a cleaner solution